message-transport/pkg/transport/metrics/receiver.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018-Present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package metrics
+
+import (
+	"github.com/projectriff/riff/message-transport/pkg/transport"
+	"github.com/projectriff/riff/message-transport/pkg/message"
+	"log"
+	"encoding/json"
+)
+
+func NewReceiver(consumer transport.Consumer) (*metricsReceiver) {
+	producerMetricsChan := make(chan ProducerAggregateMetric)
+	consumerMetricsChan := make(chan ConsumerAggregateMetric)
+
+	go func() {
+		for {
+			msg, _, err := consumer.Receive()
+			if err != nil {
+				close(consumerMetricsChan)
+				close(producerMetricsChan)
+				break
+			}
+			pm, cm := demarshallMetricMessage(msg)
```

Comment:
shouldn't that be unmarshall, seems more common?

Reply:
Yes. Renamed.